packages/database/src/modelClasses/MapOverlayGroupRelation.js

```diff
@@ -57,4 +58,72 @@ export class MapOverlayGroupRelationModel extends DatabaseModel {
       'map_overlay_group_id',
     );
   }
+
+  async find(criteria, dbOptions = {}) {
+    const options = dbOptions;
+
+    // Left join with map_overlay and map_overlay_group
+    options.multiJoin = [
+      {
+        joinType: JOIN_TYPES.LEFT,
+        joinWith: RECORDS.MAP_OVERLAY,
+        joinCondition: [
+          `${RECORDS.MAP_OVERLAY_GROUP_RELATION}.child_id`,
+          `${RECORDS.MAP_OVERLAY}.id`,
+        ],
+      },
+      {
+        joinType: JOIN_TYPES.LEFT,
+        joinWith: RECORDS.MAP_OVERLAY_GROUP,
+        joinCondition: [
+          `${RECORDS.MAP_OVERLAY_GROUP_RELATION}.child_id`,
+          `${RECORDS.MAP_OVERLAY_GROUP}.id`,
+        ],
+      },
+    ];
+
+    // Add child code for both child type options
+    options.columns = [];
+    for (const { joinWith } of options.multiJoin) {
+      const column = `${joinWith}.code as ${
+        joinWith === RECORDS.MAP_OVERLAY ? 'childMapOverlayCode' : 'childMapOverlayGroupCode'
+      }`;
+      options.columns.push(column);
+    }
+
+    // Add original field names
+    const fieldNames = await this.fetchFieldNames();
+    for (const field of fieldNames) {
+      const column = `${RECORDS.MAP_OVERLAY_GROUP_RELATION}.${field} as ${field}`;
+      options.columns.push(column);
+    }
+
+    const records = await this.database.find(this.databaseRecord, criteria, options);
+
+    const recordsWithChildCode = await Promise.all(
+      records.map(record => this.generateInstance(record)),
```

Comment:
For some reason, using

```ts
records.map(this.generateInstance)
```

uses the version of `generateInstance` from DatabaseModel, not the one one in this file (line 112) which overrides it 🤔